compiler/prelude/jsmapping.go

```diff
@@ -114,6 +114,9 @@ var $externalize = function(v, t) {
         return searchJsObject(v.$get(), t.elem);
       case $kindStruct:
         var f = t.fields[0];
+        if (f === undefined) {
```

Comment:
Can we make that `t.fields.length === 0`? That captures the intention a bit better in my regard.

Reply:
> but the defensive coder in me, that says not to trust user input

As far as I can tell, `fields` is not user input, it's a property that is generated by the GopherJS compiler. So it should be safe to rely on it existing.

If it always exists (I haven't confirmed, but that's my understanding), then `t.fields.length === 0` would be better because it captures the intention better (as Richard said) and makes it easier to read the code.

If it doesn't always exist, we should consider either making it always exist, even if it's empty for empty structs, or consider going with `if t.fields === undefined` or so.

Basically, I suggest we take the least amount of precautions that are needed to fix the issue.

We can add a test to make sure the behavior doesn't regress. Having more readable code will help with future development and maintenance. What do you think @MJKWoolnough?